tests/frontend/test-class-front-end-page-type.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * @package WPSEO\Tests\Frontend
+ */
+
+/**
+ * Unit Test Class.
+ */
+class WPSEO_Frontend_Page_Type_Test extends WPSEO_UnitTestCase {
+
+	/**
+	 * Tests the situation where nothing has been done to manipulate the result.
+	 *
+	 * @covers WPSEO_Frontend_Page_Type::is_singular()
+	 * @covers WPSEO_Frontend_Page_Type::get_singular_id()
+	 */
+	public function test_default_state() {
+		$instance = new WPSEO_Frontend_Page_Type( new WPSEO_WooCommerce_Shop_Page() );
+		$this->assertFalse( $instance->is_singular() );
+		$this->assertEquals( 0, $instance->get_singular_id() );
+	}
+
+	/**
+	 * Tests the situation where a single post will be visited.
+	 *
+	 * @covers WPSEO_Frontend_Page_Type::is_singular()
+	 * @covers WPSEO_Frontend_Page_Type::get_singular_id()
+	 */
+	public function test_singular_page() {
+		$post = $this->factory()->post->create_and_get();
+
+		$this->go_to( get_permalink( $post ) );
+
+		$instance = new WPSEO_Frontend_Page_Type( new WPSEO_WooCommerce_Shop_Page() );
```

Comment:
I would call `$instance` something more meaning full. Either `$subject` if you want to stay abstract. Or `$frontend_page_type` or `$page_type` if you want to be more concrete.